include/crm/cluster/internal.h

```diff
@@ -159,8 +159,8 @@ typedef struct pcmk__node_status {
      * void *user_data here instead, to abstract the cluster layer further.
      */
 
-    // Only used by controller
-    enum crm_join_phase join;
+    void *user_data;            //!< Arbitrary data
```

Comment:
"must be freeable by free()"

fine for now, but for more complex data we'd need to pass a free func at cluster creation